arbnode/dataposter/data_poster.go

```diff
@@ -636,20 +685,33 @@ type DataPosterConfig struct {
 	ReplacementTimes string                     `koanf:"replacement-times"`
 	// This is forcibly disabled if the parent chain is an Arbitrum chain,
 	// so you should probably use DataPoster's waitForL1Finality method instead of reading this field directly.
-	WaitForL1Finality      bool            `koanf:"wait-for-l1-finality" reload:"hot"`
-	MaxMempoolTransactions uint64          `koanf:"max-mempool-transactions" reload:"hot"`
-	MaxQueuedTransactions  int             `koanf:"max-queued-transactions" reload:"hot"`
-	TargetPriceGwei        float64         `koanf:"target-price-gwei" reload:"hot"`
-	UrgencyGwei            float64         `koanf:"urgency-gwei" reload:"hot"`
-	MinFeeCapGwei          float64         `koanf:"min-fee-cap-gwei" reload:"hot"`
-	MinTipCapGwei          float64         `koanf:"min-tip-cap-gwei" reload:"hot"`
-	MaxTipCapGwei          float64         `koanf:"max-tip-cap-gwei" reload:"hot"`
-	NonceRbfSoftConfs      uint64          `koanf:"nonce-rbf-soft-confs" reload:"hot"`
-	AllocateMempoolBalance bool            `koanf:"allocate-mempool-balance" reload:"hot"`
-	UseDBStorage           bool            `koanf:"use-db-storage"`
-	UseNoOpStorage         bool            `koanf:"use-noop-storage"`
-	LegacyStorageEncoding  bool            `koanf:"legacy-storage-encoding" reload:"hot"`
-	Dangerous              DangerousConfig `koanf:"dangerous"`
+	WaitForL1Finality      bool              `koanf:"wait-for-l1-finality" reload:"hot"`
+	MaxMempoolTransactions uint64            `koanf:"max-mempool-transactions" reload:"hot"`
+	MaxQueuedTransactions  int               `koanf:"max-queued-transactions" reload:"hot"`
+	TargetPriceGwei        float64           `koanf:"target-price-gwei" reload:"hot"`
+	UrgencyGwei            float64           `koanf:"urgency-gwei" reload:"hot"`
+	MinFeeCapGwei          float64           `koanf:"min-fee-cap-gwei" reload:"hot"`
+	MinTipCapGwei          float64           `koanf:"min-tip-cap-gwei" reload:"hot"`
+	MaxTipCapGwei          float64           `koanf:"max-tip-cap-gwei" reload:"hot"`
+	NonceRbfSoftConfs      uint64            `koanf:"nonce-rbf-soft-confs" reload:"hot"`
+	AllocateMempoolBalance bool              `koanf:"allocate-mempool-balance" reload:"hot"`
+	UseDBStorage           bool              `koanf:"use-db-storage"`
+	UseNoOpStorage         bool              `koanf:"use-noop-storage"`
+	LegacyStorageEncoding  bool              `koanf:"legacy-storage-encoding" reload:"hot"`
+	Dangerous              DangerousConfig   `koanf:"dangerous"`
+	ExternalSigner         ExternalSignerCfg `koanf:"external-signer"`
+}
+
+type ExternalSignerCfg struct {
+	// If enabled, this overrides transaction options and uses external signer
+	// for signing transactions.
+	Enabled bool `koanf:"enabled"`
```

Comment:
How about we enable the external signer when the URL is set at all?

Reply:
Sure, sounds good, changed to that.